portal/server/routes/contract/eligibility/index.js

```diff
@@ -132,120 +110,192 @@ router.get('/contract/:_id/eligibility/supporting-documentation', provide([DEAL]
 
   const { eligibility, supportingInformation = {} } = deal;
 
-  const allEligibilityValidationErrors = mergeEligibilityValidationErrors(
-    deal.eligibility,
-    deal.supportingInformation,
-  );
+  const allEligibilityValidationErrors = mergeEligibilityValidationErrors(deal.eligibility, deal.supportingInformation);
   const validationErrors = generateErrorSummary(allEligibilityValidationErrors, errorHref);
 
   const documentationValidationErrors = generateErrorSummary(supportingInformation.validationErrors, eligibilityErrorHref);
 
   const completedForms = completedEligibilityForms(deal.eligibility.status, validationErrors);
 
-  return res.render(
-    'eligibility/eligibility-supporting-documentation.njk',
-    {
-      _id: deal._id,
-      supportingInformation,
-      eligibility,
-      validationErrors: documentationValidationErrors,
-      additionalRefName: deal.additionalRefName,
-      user: req.session.user,
-      taskListItems: eligibilityTaskList(completedForms),
-    },
-  );
+  return res.render('eligibility/eligibility-supporting-documentation.njk', {
+    _id: deal._id,
+    supportingInformation,
+    eligibility,
+    validationErrors: documentationValidationErrors,
+    additionalRefName: deal.additionalRefName,
+    user: req.session.user,
+    taskListItems: eligibilityTaskList(completedForms),
+  });
 });
 
-router.post('/contract/:_id/eligibility/supporting-documentation', upload.any(), async (req, res) => {
-  const { _id, userToken } = requestParams(req);
-  const { body, files, query } = req;
-  const formData = { ...body };
-
-  if (query.removefile) {
-    formData.deleteFile = query.removefile;
-  }
-
-  const updatedDeal = await getApiData(
-    api.updateEligibilityDocumentation(_id, formData, files, userToken),
-    res,
-  );
+const renderSupportingDocumentationPageWithUploadErrors = (uploadError, deal, req, res) => {
```

Comment:
This new function is called when the multer filter has populated `res.locals`, a common place to store persisting information in an express app. It formats the errors from the multer filter to return in the njk template against the correct upload field